### PR TITLE
Add mention to RFC 940 in the Rust Reference.

### PR DESCRIPTION
**Pull request by andradei**
This PR adds a mention to hyphens in Cargo package names being replaced by underscores when used as a crate, as per [RFC 940](https://github.com/rust-lang/rfcs/blob/master/text/0940-hyphens-considered-harmful.md)

It also formats the RFCs consistently as RFC XXX instead of RFCXXX.

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -845,6 +845,20 @@ extern crate std; // equivalent to: extern crate std as std;
 extern crate std as ruststd; // linking to 'std' under another name
 ```
 
+When naming Rust crates, hyphens are disallowed. However, Cargo packages may
+make use of them. In such case, when `cargo.toml` doesn't specify a crate name,
```
Comment (Manishearth) [inline src/doc/reference.md]: Cargo.toml is has a capital c

**Comment by Manishearth** [comment]
lgtm, one nit

File: src/doc/reference.md [hunk 2]
```diff
@@ -845,6 +845,20 @@ extern crate std; // equivalent to: extern crate std as std;
...
+Cargo will transparently replace `-` with `_` (Refer to [RFC 940] for more
+details).
+
+Here is an example:
+
+```{.ignore}
+// Importing the Cargo package hello-world/
+extern crate hello_world; // The crate name replaced the hyphen
```
Comment (birkenfeld) [inline src/doc/reference.md]: * trailing `/` looks suspicious?
* did you mean "The underscore replaced the hyphen"?

**Comment by andradei** [comment]
Thanks for catching the discrepancies. I hope the changes are better phrased now.

**Comment by apasel422** [comment]
Ping @Manishearth.

**Pull request merged**